File: java/liskov-refactor/src/test/java/no/bouvet/solid/lsp/BirdServiceTest.java
package no.bouvet.solid.lsp;

import java.util.Arrays;
import java.util.List;

import org.junit.Before;
import org.junit.Test;

public class BirdServiceTest {
    private BirdService subject;

    @Before
    public void setUp() {
        subject = new BirdService();
    }

    @Test
    public void testFeedThemAll() {
        subject.feedThemAll(allBirds());
    }

    @Test
    public void testLetThemGo() {
        subject.letThemGo(allFlyingBirds());
    }

    private List<Bird> allBirds() {
        return Arrays.asList(new Duck(), new Penguin(), new Crow());
    }

    private List<FlyingBird> allFlyingBirds() {
        return Arrays.asList(new Duck(), new Crow());
    }

}

File: dotnet/Solid/RefactoredTests/LSP/Example1/BirdServiceTest.cs
using System.Collections.Generic;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using RefactoredSolid.LSP.Example1;

namespace RefactoredTests.LSP.Example1
{
    [TestClass]
    public class BirdServiceTest
    {
        private BirdService _subject;


        [TestInitialize]
        public void Setup()
        {
            _subject = new BirdService();
        }

        [TestMethod]
        public void TestFeedThemAll()
        {
            _subject.FeedThemAll(AllBirds());
        }

        [TestMethod]
        public void TestLetThemGo()
        {
            _subject.LetThemGo(AllFlyingBirds());
        }

        private List<Bird> AllBirds()
        {
            return new List<Bird>
                   {
                        new Duck(),
                        new Penguin(),
                        new Crow()
                   };
        }

        private List<FlyingBird> AllFlyingBirds()
        {
            return new List<FlyingBird>
                   {
                        new Duck(),
                        new Crow()
                   };
        }
    }
}
